shuffle/README.md

```diff
@@ -0,0 +1,32 @@
+# Fisher-Yates Shuffle function test vectors
+
+We offer test vectors for the implementation of the shuffle function as described in Appendix F of the Gray Paper.
+This section presents two definitions, a recursive one that uses an integer sequence as random seeds in Eq 329, and one (Eq 331) which takes a 32-byte hash as entropy, and is defined in terms of the previous one.
+
+The implementation of Eq 329 is not generally used in the Gray Paper, and so we present test vectors only for the version of Eq 331. However, the present script creates these test cases with recourse to the Eq 329 implementation, to reproduce exactly the definition of the Gray Paper.
+
+## Notes about test parameters
+
+In every case, the output must be a permutation of the input sequence, and this is checked by the generating script.
+In the GP, at the moment, this function is used only to shuffle sequences of validators or cores, both of which are identified by integers. For that reason, all test cases use integers as the input type. All the input elements are distinct to ensure the final positions of each element are unambiguously determined.
+
+All test cases use the same entropy, which is a 32-byte sequence with values from 0 to 31.
```

Comment:
I didn't notice this. Is there a reason to use the same entropy for each test? Would seem like a good idea to test a few different ones!

Reply:
No reason at all, it just made the coding and verification simpler. 
But you're right, adding different entropies makes the tests stronger. I'll fix that.